### PR TITLE
This PR updates contract interaction

### PR DESCRIPTION
**Pull request by yathishram**
### Description:
This PR contains contract interaction updates

### **Added**

- The following APIs and features are added to this PR

Makes abi and address access for contracts dynamic

Core APIs

- [x] syncSafe - Adds offChain data to onChain

### **Pending**

- [x] Adding giveRuling API for testing purpose
- [x] Appropriate data types at some places
- [ ] Removing contract from the `@safient/claims` and creating a new directory structure

### CODE REVIEW

File: src/contracts/SafientMain.ts [hunk 1]
```diff
@@ -1,12 +1,13 @@
 import { Safe, Claim, Tx, ContractAddress, ContractABI, Signer, RecoveryProof } from '../types/Types';
-import { safientMainABI } from '../abis/SafientMain';
 import { Contract } from '@ethersproject/contracts';
 import { BigNumber } from '@ethersproject/bignumber';
 import { formatEther } from '@ethersproject/units';
 import { Logger } from '@ethersproject/logger';
 import { Bytes } from 'ethers';
 import networks from '../utils/networks.json';
-
+import {artifacts} from 'hardhat'
```
Comment (koshikraj) [inline src/contracts/SafientMain.ts:8]: Remove the dependency if not used.

File: src/contracts/Arbitrator.ts [hunk 2]
```diff
@@ -1,10 +1,12 @@
-import { arbitratorABI } from '../abis/Arbitrator';
 import { ContractAddress, ContractABI, Signer } from '../types/Types';
 import { Contract } from '@ethersproject/contracts';
 import { BigNumber } from '@ethersproject/bignumber';
 import { formatEther } from '@ethersproject/units';
 import { Logger } from '@ethersproject/logger';
 import networks from '../utils/networks.json';
+import { artifacts } from 'hardhat';
```
Comment (koshikraj) [inline src/contracts/Arbitrator.ts:7]: Remove the dependency if not used.

File: scripts/deploy.js [hunk 3]
```diff
@@ -17,9 +17,15 @@ const deploy = async (contractName, _args = [], overrides = {}, libraries = {})
   const contractArgs = _args || [];
   const contractArtifacts = await hre.ethers.getContractFactory(contractName, { libraries: libraries });
   const deployed = await contractArtifacts.deploy(...contractArgs, overrides);
+  const data = await hre.artifacts.readArtifact(contractName);
   const encoded = abiEncodeArgs(deployed, contractArgs);
 
   fs.writeFileSync(`artifacts/${contractName}.address`, deployed.address);
+  const deployedAddress = {
```
Comment (koshikraj) [inline scripts/deploy.js:24]: Change the variable name

File: contracts/SafientMain.sol [hunk 4]
```diff
@@ -217,6 +245,36 @@ contract SafientMain is IArbitrable, IEvidence {
         emit CreateSafe(msg.sender, _inheritor, metaEvidenceID);
     }
 
+    function syncSafe(
+        address _creator,
+        string memory _safeId,
+        string calldata _metaEvidence
+    ) external payable syncSafeRequisite(_creator, _safeId, _metaEvidence) {
+        metaEvidenceID += 1;
+
```
Comment (koshikraj) [inline contracts/SafientMain.sol:254]: The safe creation proof part is yet to be worked on, right?

Comment (yathishram) [reply contracts/SafientMain.sol:254]: Yeah, I wanted to discuss that once before going forward. I was planning to do it in the next release.

File: src/contracts/Arbitrator.ts [hunk 5]
```diff
@@ -54,4 +56,10 @@ export class Arbitrator {
       this.logger.throwError(e.message);
     }
   };
+
+  givingRulingCall = async(disuputeId:number, ruling:number): Promise<any> => {
```
Comment (koshikraj) [inline src/contracts/Arbitrator.ts:60]: Update the function name to `giveRulingCall`

Comment (yathishram) [reply src/contracts/Arbitrator.ts:60]: Yeah, @devpavan04 is finishing it up

**Comment by koshikraj** [comment]
LGTM

**Pull request merged**